tests/cross/src/test/scala/tests/tokens/SemanticTokensScala3Suite.scala

```diff
@@ -9,23 +9,24 @@ class SemanticTokensScala3Suite extends BaseSemanticTokensSuite {
 
   check(
     "enum",
-    s"""|package <<example>>/*namespace*/
-        |
-        |enum <<FooEnum>>/*enum,abstract*/:
-        |  case <<Bar>>/*enum*/, <<Baz>>/*enum*/
-        |object <<FooEnum>>/*class*/
-        |""".stripMargin,
+    """|package <<example>>/*namespace*/
+       |
+       |enum <<FooEnum>>/*enum,abstract*/:
+       |  case <<Bar>>/*enum*/, <<Baz>>/*enum*/
+       |object <<FooEnum>>/*class*/
+       |""".stripMargin,
   )
 
   check(
     "enum1",
-    s"""|package <<example>>/*namespace*/
-        |
-        |enum <<FooEnum>>/*enum,abstract*/:
-        |  case <<A>>/*enum*/(<<a>>/*variable,readonly*/: <<Int>>/*class,abstract*/)
-        |  case <<B>>/*enum*/(<<a>>/*variable,readonly*/: <<Int>>/*class,abstract*/, <<b>>/*variable,readonly*/: <<Int>>/*class,abstract*/)
-        |  case <<C>>/*enum*/(<<a>>/*variable,readonly*/: <<Int>>/*class,abstract*/, <<b>>/*variable,readonly*/: <<Int>>/*class,abstract*/, <<c>>/*variable,readonly*/: <<Int>>/*class,abstract*/)
-        |""".stripMargin,
+    """|package <<example>>/*namespace*/
+       |
+       |enum <<FooEnum>>/*enum,abstract*/:
+       |  case <<A>>/*enum*/(<<a>>/*variable,declaration,readonly*/: <<Int>>/*class,abstract*/)
+       |  case <<B>>/*enum*/(<<a>>/*variable,declaration,readonly*/: <<Int>>/*class,abstract*/, <<b>>/*variable,declaration,readonly*/: <<Int>>/*class,abstract*/)
+       |  case <<C>>/*enum*/(<<a>>/*variable,declaration,readonly*/: <<Int>>/*class,abstract*/, <<b>>/*variable,declaration,readonly*/: <<Int>>/*class,abstract*/, <<c>>/*variable,declaration,readonly*/: <<Int>>/*class,abstract*/)
+       |
+       |""".stripMargin,
```

Comment:
Let's add this test
```scala
check(
      "issue",
      s"""|package <<example>>/*namespace*/
          |
          |object <<A>>/*class*/ {
          |  val <<x>>/*variable,definition,readonly*/ = <<List>>/*variable,readonly*/(1,2,3)
          |  val <<s>>/*variable,definition,readonly*/ = <<Some>>/*class*/(1)
          |  val <<Some>>/*class*/(<<s1>>/*variable,declaration,readonly*/) = <<s>>/*variable,readonly*/
          |  val <<Some>>/*class*/(<<s2>>/*variable,declaration,readonly*/) = <<s>>/*variable,readonly*/
          |}
          |""".stripMargin,
    )
```
It looks nondeterministic, sometimes `s2` is `definition` and `s1` is `declaration`, sometimes both are `definition` or `declaration`. Removing definition of `x` also changes the behaviour

Reply:
Ok, it seems we get two nodes in this case. One from Bind and the other from Definition and both point to the same identifier :thinking: 

The easiest might be to make Bind also a definition and we can interpret it that the value cannot change, so maybe we can make `declaration` much narrower. Since it's not clear how to interpret these values we can say declartion is only if right hand side is empty.